runtime/bcverify/rtverify.c

```diff
@@ -585,6 +586,15 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
 					/* Modify the liveStack temporarily to contain the handler exception */
 					if (handler->exceptionClassIndex) {
 						catchName = J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *)(&constantPool [handler->exceptionClassIndex]));
+#if JAVA_SPEC_VERSION >= 17
+						if (NULL == j9rtv_verifierGetRAMClass(verifyData, verifyData->classLoader, J9UTF8_DATA(catchName), J9UTF8_LENGTH(catchName), &reasonCode)) {
+							if (BCV_ERR_INSUFFICIENT_MEMORY == reasonCode) {
+								goto _outOfMemoryError;
+							} else {
+								goto _noClassDefError;
+							}
+						}
+#endif /* JAVA_SPEC_VERSION >= 17 */
```

Comment:
This will process the handlers more than once - would it be better to handle this either up front in this method or in something like `staticverify.c::checkMethodStructure()`?

Reply:
The only question is whether the check is only focusing on an empty class which can be done in the static verifier; otherwise, a check in the runtime verifier is expected to achieve this.

Reply:
> would it be better to handle this either up front in this method or in something like staticverify.c::checkMethodStructure()?

Yes as Cheng said, we need to be able to perform a classload in order to validate if the catch class is defined or define-able.

Reply:
Immediately after calling `verifyBytecodes()`, we call `verifyExceptions()` which checks that the exception handlers actually name subclasses of Throwable.  For many cases, this already forces the class to be loaded at that point.

https://github.com/eclipse-openj9/openj9/blob/fdeac8f10f26a2532513534d3c526af07f933684/runtime/bcverify/rtverify.c#L2531-L2533

Can we add this check to `verifyExceptions()` so it occurs once per-handler range?  Loading the class to ensure it's accessible can also speed up the `isSubclass(Throwable, x)` check